doc/docusaurus/docs/working-with-scripts/script-purposes.md

```diff
@@ -0,0 +1,80 @@
+---
+sidebar_position: 2
+---
+
+# Script Purposes
+
+One of the arguments every Plutus script receives is the script context, containing information about the transaction the script is validating.
+One of the fields of script context is the script purpose.
+Since a transaction may do multiple things, each of which is validated by a separate  script, the script purpose informs a script what exactly it is supposed to validate.
+
+Plutus V1 and V2 share the same [`ScriptPurpose`](https://plutus.cardano.intersectmbo.org/haddock/master/plutus-ledger-api/PlutusLedgerApi-V1-Contexts.html#t:ScriptPurpose) type with four variants: spending, minting, rewarding and certifying.
+Plutus V3's [`ScriptPurpose`](https://plutus.cardano.intersectmbo.org/haddock/master/plutus-ledger-api/PlutusLedgerApi-V3-Contexts.html#t:ScriptPurpose) has two additional variants: voting and proposing.
+Next we go over and explain all these different script purposes.
+
+## Spending
+
+A spending script validates the spending of a UTXO.
+A UTXO can have either a public key address or a script address.
+To spend a UTXO with a script address, the script whose hash matches the script address must be included in the transaction (either directly, or as a reference script), and is executed to validate it.
+
+The script can make the decision based on the datum attached in the UTXO being spent, if any (datum is mandatory for Plutus V1 and V2, i.e., a UTXO with a Plutus V1 or V2 script address but without a datum cannot be spent; datum is optional for Plutus V3), the redeemer included in the transaction, as well as other fields of the transaction.
+
+The `Spending` constructor of `ScriptPurpose` contains a `TxOutRef` field, which is the UTXO the transaction is attempting to spend.
```

Comment:
I think this is a bit clearer (just a suggestion):

"The `Spending` constructor of `ScriptPurpose` includes a `TxOutRef` field that references the UTXO the script is validating, which is also one of the UTXOs the transaction attempts to spend."